core/src/test/scala/org/typelevel/paiges/PaigesScalacheckTest.scala

```diff
@@ -212,50 +212,21 @@ class PaigesScalacheckTest extends FunSuite {
   }
 
   test("the left and right side of a union are the same after flattening") {
-    import Doc._
-    def okay(d: Doc): Boolean = d match {
-      case Empty | Text(_) | Line(_) => true
-      case Concat(a, b) => okay(a) && okay(b)
-      case Nest(j, d) => okay(d)
-      case Align(d) => okay(d)
-      case LazyDoc(d) => okay(d.evaluated)
-      case Union(a, b) =>
-        (a.flatten === b.flatten) && okay(a) && okay(b)
+    forAll { u: Doc.Union => assert(u.a.flatten === u.b.flatten) }
+  }
+
+  test("the left side of a union has a first line as long or longer than the right") {
+    forAll(Gen.choose(1, 200), genUnion) { (n, u) =>
+      if (u.nonEmpty) {
+        def firstLine(d: Doc) = {
```

Comment:
@oscar-stripe I decided to do the fix here, rather than replace the existing function, since I don't think ignoring the rhs of `Union` is correct.  If I'm mistaken please let me know and I'll happily replace the existing loop.